### PR TITLE
Fix front js interface bug

### PR DESCRIPTION
**Pull request by Enc3phale**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | N
| Related developer documentation PR URL | N
| Issues addressed (#s or URLs) | N
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR corrects some interface bugs

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a segment
2. First bug:
Add first filter -> first line is cut and shifted

screenshot_2018-11-19 edit segment - segment bug interface mautic 1

3. Second bug:
Add an other filter and remove the first -> the new first line is cut, shifted and glue won't disappear.

screenshot_2018-11-19 edit segment - segment bug interface mautic 2

4. Third bug
Add filter `Segment Membership` with including filter, chose some value, drag&drop the line and save ->selected value are not saved and the filter is reset to empty

screenshot_2018-11-19 edit segment - segment bug interface mautic 3

#### Steps to test this PR:
1. Apply PR
2. Regenerate assets
3. Clear browser cache
4. Repeat steps above
5. All is good

### CODE REVIEW

**Comment by alanhartless** [comment]
@Enc3phale this works for editing existing segments. But it doesn't seem to work for new segments.

**Comment by Enc3phale** [comment]
@alanhartless I have fixed this

File: app/bundles/LeadBundle/Assets/js/lead.js [hunk 1]
```diff
@@ -389,7 +390,7 @@ Mautic.reorderSegmentFilters = function() {
             }
 
             var newName = prefix+'[filters]['+counter+']['+suffix+']';
-            if (name.slice(-2) === '[]') {
+            if (name !== undefined && name.slice(-2) === '[]') {
```
Comment (escopecz) [inline app/bundles/LeadBundle/Assets/js/lead.js]:
```suggestion
            if (typeof name !== 'undefined' && name.slice(-2) === '[]') {
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined with the section called "DON'T DO THIS"

Comment (Enc3phale) [reply app/bundles/LeadBundle/Assets/js/lead.js]: "DON'T DO THIS" is about scope of undefined.

`name !== undefined` and `typeof name !== 'undefined'` is exactly same (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#Strict_equality_and_undefined)

The one reason to use typeof is that it does not throw an error if the variable has not been declared, and it's declared L. 385

Comment (escopecz) [reply app/bundles/LeadBundle/Assets/js/lead.js]: The problem is that `undefined` is not a reserved keyword. So anyone can define a variable called `undefined` in a parent scope and break your code.

**Pull request merged**